spec/test_data/v5.1/expected_headmaster.rbi

```diff
@@ -0,0 +1,681 @@
+# This is an autogenerated file for dynamic methods in Headmaster
+# Please rerun bundle exec rake rails_rbi:models[Headmaster] to regenerate.
+
+# typed: strong
+module Headmaster::ActiveRelation_WhereNot
+  sig { params(opts: T.untyped, rest: T.untyped).returns(T.self_type) }
+  def not(opts, *rest); end
+end
+
+module Headmaster::GeneratedAttributeMethods
+  extend T::Sig
+
+  sig { returns(Integer) }
+  def id; end
+
+  sig { params(value: T.any(Numeric, ActiveSupport::Duration)).void }
+  def id=(value); end
+
+  sig { returns(T::Boolean) }
+  def id?; end
+
+  sig { returns(T.nilable(Integer)) }
```

Comment:
I thought this attribute has validation and should use non-nilable value as return type?

Reply:
Oh, this is interesting. In the example `school` has a presence validation and that `belongs_to` logic is happening in the `ActiveRecordAssoc`. If the association were not nilable then theoretically the FK should also be not nilable but we don't consider the association in the `ActiveRecordAttribute` plugin.

This would be the same behavior as the existing logic around `belongs_to_and_required` since that's only handled in `ActiveRecordAssoc` plugin, not the attribute one. Given that, would you be OK not handling this in the current PR and I could maybe follow-up with something that would update `ActiveRecordAttribute` to inspect associations (i.e. look at validations but also optional/required)?